documentation/boltspec_reference.md

```diff
@@ -1225,7 +1227,562 @@ The `allow_upload` function accepts the following modifiers:
+  ```
+
+- `always_return(value)`
+
+  Sets the value for each target's `Result` for any task that is run. Returns a
+  `Bolt::ResultSet` object.
+
+  ```ruby
+  allow_any_task.always_return({ 'happy' => true })
+  ```
+
+- `return_for_targets(targets_to_values)`
+
+  Sets the value for each target's `Result` when any task is run. Accepts a
+  hash of key-value pairs where each key is a target and the value is the value
+  for that target's `Result`.
+
+  ```ruby
+  allow_any_task('pet_dog').return_for_targets(
+    'target1' => { 'happy' => true },
+    'target2' => { 'happy' => false }
+  )
+  ```
+
+- `return(&block) { |targets, task, params| ... }`
+
+  Invokes a block to construct the `ResultSet` returned by the `run_task`
+  function.
+
+  ```ruby
+  allow_any_task.return do |targets, task, params|
+    results = targets.map do |target|
+      Bolt::Result.new(target, value: { 'happy' => true })
+    end
+
+    Bolt::ResultSet.new(results)
+  end
+  ```
+
+### `allow_any_upload`
+
+The `allow_upload` function stubs any invocation of the [`upload_file`
+function](plan_functions.md#upload-file). It does not accept parameters.
+
+```ruby
+allow_any_upload
+```
+
+The `allow_any_upload` function accepts the following modifiers:
+
+- `be_called_times(number)`
+
+  The test fails if any one file is uploaded more than _number_ times.
+
+  ```ruby
+  allow_any_upload.be_called_times(1)
+  ```
+
+- `not_be_called`
+
+  The test fails if any file is uploaded.
+
+  ```ruby
+  allow_any_upload.not_be_called
+  ```
+
+- `with_targets(targets)`
+
+  The target or list of targets that a file can be uploaded to. The test
+  fails if any file is uploaded to a different list of targets.
+
+  ```ruby
+  allow_any_upload.with_targets(['target1', 'target2', 'target3'])
+  ```
+
+- `with_params(parameters)`
+
+  The [options](plan_functions.md#upload-file) that can be passed to the
+  `upload_file` function. The test fails if any file is uploaded with a
+  different set of  options.
+
+  ```ruby
+  allow_any_upload.with_params({ '_run_as' => 'root' })
+  ```
+
+- `with_destination(destination)`
+
+  The destination path that a file is uploaded to. The test fails if any
+  file is uploaded to a different location.
+
+  ```ruby
+  allow_any_upload.with_destination('/etc/ssh/sshd_config')
+  ```
+
+- `return(&block) { |targets, source, destination, params| ... }`
+
+  Invokes a block to construct the `ResultSet` returned by the `upload_file`
+  function.
+
+  ```ruby
+  allow_any_upload.return do |targets, source, destination, params|
+    results = targets.map do |target|
+      Bolt::Result.new(target, value: { 'path' => File.join(destination, source) })
+    end
+
+    Bolt::ResultSet.new(results)
+  end
+  ```
+
+- `error_with(error)`
+
+  Sets the error hash for each target's `Result` when any file is uploaded.
+  Returns a `Bolt::ResultSet` object.
+
+  ```ruby
+  allow_any_upload.error_with('msg' => 'Not authorized')
+  ```
+
+### `allow_any_out_verbose`
```

Comment:
The `allow_any_out_verbose` section should come immediately after the `allow_any_out_message` section.